### ISSUE TITLE
Troubleshoot RPC calls for tranche prices

### ISSUE THREAD

**Issue by filo87** [embrio-tech/centrifuge-subql#39]
### Summary

RPC calls are currently not decorated in the subql node

### Steps to reproduce
include RPC chaintypes in `chaintypes.ts` and use the corresponding `api.rpc.<pallet>.<method>(...)` call

### Current behavior

Subql reports:

```bash
api.rpc.*.*() Not permitted
```

### Expected behavior

RPC calls should be feasible

**Comment by filo87** [comment]
### Solution

Custom RPC methods have strict requirements to work with SubQL. 
RPC calls are supported by subquery ONLY if they are of `isHistoric` type.

This means one argument of the RPC method (usually the last one) should be a blockHash where the chain is queried,
this is the only way to allow subql to query the RPC at a specific block, and not always get the latest result.

Therefore the RPC method we use must have a structure like this: 

```
api.rpc.pools.trancheTokenPrices(pool_id, at?)
```

and we need to implement it accordingly in subquery.

basically it is mentioned also here: https://doc.subquery.network/build/mapping.html#custom-chain-rpc-calls

some default RPC methods already have this structure like for example this here: 

https://polkadot.js.org/docs/substrate/rpc/#getreadproofkeys-vecstoragekey-at-blockhash-readproof
